### PR TITLE
Use CometBFT 0.37 compatibility mode for unsupported Tendermint versions (eg. Tendermint 0.35 and 0.36)

### PR DESCRIPTION
**Pull request by seanchen1991**
Closes: #XXX

## Description

Downgrades the Tendermint version check to a warning instead of an error. 

______

### PR author checklist:
- [ ] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests: integration (for Hermes) or unit/mock tests (for modules).
- [ ] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).
- [ ] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: crates/relayer/src/chain/cosmos.rs [hunk 1]
```diff
@@ -850,8 +850,10 @@ impl ChainEndpoint for CosmosSdkChain {
 
         let node_info = rt.block_on(fetch_node_info(&rpc_client, &config))?;
 
-        let compat_mode = CompatMode::from_version(node_info.version)
-            .map_err(|e| Error::rpc(config.rpc_addr.clone(), e))?;
+        let compat_mode = CompatMode::from_version(node_info.version).unwrap_or_else(|e| {
+            warn!("Unsupported tendermint version, will use v0.347 compatibility mode but relaying might not work as desired: {e}");
```
Comment (romac) [inline crates/relayer/src/chain/cosmos.rs:854]:
```suggestion
            warn!("Unsupported tendermint version, will use v0.37 compatibility mode but relaying might not work as desired: {e}");
```

**Review by romac — APPROVED** [review]

**Comment by epanchee** [comment]
Hey.
Wondering does it work with Sei forked tendermint version? They currently name it `0.35.0-unreleased`.
Thanks to this PR Hermes new version will be able to relay to/from Sei.

This is their diff:  https://github.com/sei-protocol/sei-tendermint/compare/46d0a598a7f5c67cbdefea37c8da18df2c25d184...v0.2.21

Not sure which version they forked from.

**Comment by seanchen1991** [comment]
> Wondering does it work with Sei forked tendermint version?

Just to be clear, your concern is that this branch of Hermes doesn't work with Sei's forked version of Tendermint, right? It's not to do with whether this branch is able to relay to/from Sei chains.

**Comment by epanchee** [comment]
> > Wondering does it work with Sei forked tendermint version?
> 
> Just to be clear, your concern is that this branch of Hermes doesn't work with Sei's forked version of Tendermint, right? It's not to do with whether this branch is able to relay to/from Sei chains.

Yes. Built Hermes from `master` branch and saw the error above

**Comment by romac** [comment]
> Update: Can confirm that Sei's tendermint version is not compatible with `CompatMode::V0_37`:
> 
> ```
> 2023-06-12T11:45:55.541879Z  WARN ThreadId(28) spawn:chain{chain=...}:client{client=07-tendermint-10}:connection{connection=connection-4}:channel{channel=channel-8}:worker.packet.cmd{src_chain=.. src_port=transfer src_channel=channel-8 dst_chain=pisco-1}:schedule_packet_clearing{height=Some(Height { revision: 1, height: 4441592 })}:relay_pending_packets{height=Some(Height { revision: 1, height: 4441591 })}:build_packet_ack_msgs{query_height=1-4441591}: encountered query failure while pulling packet data: RPC error to endpoint [...]
> 
> Caused by:
>    0: serde parse error
>    1: invalid length 0, expected struct EvidenceList with 1 element at line 1 column 1346
> ```

@epanchee Can you please re-run Hermes with the flag `--debug=rpc`? (eg. `hermes --debug=rpc start`)

This will output the JSON-RPC response that failed to parse right before the error. Can you then paste this here in this thread?

**Comment by epanchee** [comment]
@romac I ran with `--debug=rpc` and it makes sense now. I was using public RPC which has some throttle limit.

```
2023-06-27T11:57:53.646187Z DEBUG ThreadId(30) event_source.rpc{chain.id=atlantic-2}: Incoming response: <html>
<head><title>429 Too Many Requests</title></head>
<body>
<center><h1>429 Too Many Requests</h1></center>
<hr><center>nginx/1.18.0 (Ubuntu)</center>
</body>
</html>

2023-06-27T11:57:53.646305Z ERROR ThreadId(30) event_source.rpc{chain.id=atlantic-2}: event source encountered an error: RPC error: serde parse error: expected value at line 1 column 1
```

Switching to private Astroport infra solved this issue.
I'm sorry for a false alaram.

**Comment by romac** [comment]
> I'm sorry for a false alaram.

No worries!

**Pull request merged**